Assets/Scripts/GameManager.cs

```diff
@@ -75,4 +102,33 @@ private void RestartLevel()
         _UIManager.RestartLevelUIReset();
         LevelUpdate();
     }
+
+    private void Save()
+    {
+        PlayerPrefs.SetInt("savedLevel", _currentLevel);
+        PlayerPrefs.SetFloat("savedTime", _currentTime);
+        PlayerPrefs.SetInt("savedScore", _shapesManager.score);
+    }
+
+    private void Load()
+    {
+        _currentLevel = PlayerPrefs.GetInt("savedLevel");
```

Comment:
chances of these calls failing are minimal BUT it is good practice to include error handling and/or [default values](https://docs.unity3d.com/ScriptReference/PlayerPrefs.GetInt.html) for the possibility that these calls could fail. please add :)